### PR TITLE
fix(cognito): deprecate privateKey and add privateKeyValue as typed SecureValue

### PR DESCRIPTION
**Pull request by pahud**
### Issue # (if applicable)

Closes https://github.com/aws/aws-cdk/issues/31378

### Reason for this change

1. `privateKey` was typed `string` which should be `SecureValue` just as [clientSecretValue](https://github.com/aws/aws-cdk/blob/1e203753519e10e19ef0db87e1382377b609bcaa/packages/aws-cdk-lib/aws-cognito/lib/user-pool-idps/google.ts#L28) in Google IdP. This PR deprecates `privateKey` and adds `privateKeyValue` with correct type.
2. `apple.ts` was named by mistake and it won't be unit tested. This PR renames it to `apple.test.ts` so it would be covered. Figured an existing test was failed, just fixed that failed one as well.

### Description of changes

- Add `privateKeyValue` property of type SecretValue to UserPoolIdentityProviderAppleProps
- Deprecate the existing `privateKey` string property
- Implement logic to ensure exactly one of `privateKey` or `privateKeyValue` is provided
- Update UserPoolIdentityProviderApple to use the new `privateKeyValue` when available
- Rename apple.ts test file to apple.test.ts for consistency
- Add new test case to verify mutually exclusive properties

Users must now provide either `privateKey` or `privateKeyValue`,
but not both. This change enhances security by allowing the use of SecretValue
for the Apple IDP private key.

### Checklist
- [x] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md) and [DESIGN GUIDELINES](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md)

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/aws-cdk-lib/aws-cognito/test/user-pool-idps/apple.test.ts [hunk 1]
```diff
@@ -90,8 +90,8 @@ describe('UserPoolIdentityProvider', () => {
         keyId: 'CDKKEYCDK1',
         privateKey: 'PRIV_KEY_CDK',
         attributeMapping: {
-          familyName: ProviderAttribute.APPLE_LAST_NAME,
-          givenName: ProviderAttribute.APPLE_FIRST_NAME,
+          familyName: ProviderAttribute.APPLE_FIRST_NAME,
```
Comment (pahud) [inline packages/aws-cdk-lib/aws-cognito/test/user-pool-idps/apple.test.ts:93]: This is not part of this PR but it won't pass if unchanged so we have to update it. I am not sure if relevant code has to be fixed but we don't fix that in this PR.

File: packages/aws-cdk-lib/aws-cognito/lib/user-pool-idps/apple.ts [hunk 2]
```diff
@@ -44,6 +52,12 @@ export class UserPoolIdentityProviderApple extends UserPoolIdentityProviderBase
 
     const scopes = props.scopes ?? ['name'];
 
+    //at least one of the properties must be configured
```
Comment (GavinZZ) [inline packages/aws-cdk-lib/aws-cognito/lib/user-pool-idps/apple.ts:55]:
```suggestion
    // Exactly one of the properties must be configured
```

File: packages/aws-cdk-lib/aws-cognito/test/user-pool-idps/apple.test.ts (hunk 1 above)
Comment (GavinZZ) [reply packages/aws-cdk-lib/aws-cognito/test/user-pool-idps/apple.test.ts:93]: Instead of changing this, I think you should change the verification step. Family name is last name.

Comment (pahud) [reply packages/aws-cdk-lib/aws-cognito/test/user-pool-idps/apple.test.ts]: yes you are right!!

**Comment by pahud** [comment]
Exemption Request

**Comment by pahud** [comment]
@GavinZZ 

Unfortunately I don't have a valid private_key for that. And there's no existing integ test for that as well so I am offering unit tests only.

>8:07:08 PM | CREATE_FAILED        | AWS::Cognito::UserPoolIdentityProvider | AppleIdentityProvider6271AB3B
Provided private key cannot be used for Sign in with Apple. (Service: AWSCognitoIdentityProviderService; Status Code: 400; Error Code: InvalidParameterException;
Request ID: b28d3b10-14ca-42bc-a288-e1db34e7d6e3; Proxy: null)

**Review by GavinZZ — APPROVED** [review]
